lib/CXGN/BrAPI/v2/ObservationVariables.pm

```diff
@@ -124,240 +145,261 @@ sub search {
+        my $refs= CXGN::BrAPI::v2::ExternalReferences->new({
+            bcs_schema => $self->bcs_schema,
+            dbxref_id => $variable->dbxref_id
+        });
+        my $external_references = $refs->references_db();
+
         %result = (
             additionalInfo => undef,
             commonCropName => $supported_crop,
             contextOfUse => undef,
-            defaultValue => $trait->default_value,
-            documentationURL => $trait->uri,
-            externalReferences => $db_name.":".$accession,
+            defaultValue => $variable->default_value,
+            documentationURL => $variable->uri,
+            externalReferences => $external_references,
             growthStage => undef,
             institution  => undef,
             language => 'eng',
-            method => {
-                # additionalInfo
-                # bibliographicalReference
-                # description
-                # externalReferences
-                # formula
-                # methodClass
-                # methodDbId
-                # methodName
-                # ontologyReference => {
-                #         documentationLinks
-                #         ontologyDbId
-                #         ontologyName
-                #         version
-                #     }
-                },
-            observationVariableDbId => qq|$cvterm_id|,
-            observationVariableName => $cvterm_name."|".$db_name.":".$accession,
+            method => $method_json,
+            observationVariableDbId => $variable->cvterm_id,
+            observationVariableName => $variable->display_name,
             ontologyReference => {
-                documentationLinks => $db_url,
-                ontologyDbId => qq|$db_id|,
-                ontologyName => $db_name,
+                documentationLinks => $variable->uri ? $variable->uri : undef,
+                ontologyDbId => $variable->db_id ? $variable->db_id : undef,
+                ontologyName => $variable->db ? $variable->db : undef,
                 version => undef,
             },
-            scale => {
-                datatype => $trait->format,
-                decimalPlaces => undef,
-                externalReferences => '',
-                ontologyReference => {
-                #         documentationLinks
-                #         ontologyDbId
-                #         ontologyName
-                #         version
-                },
-                scaleDbId => undef,
-                scaleName => undef,
-                validValues => {
-                    min =>$trait->minimum ? $trait->minimum : undef,
-                    max =>$trait->maximum ? $trait->maximum : undef,
-                    categories => \@brapi_categories,
-                },
-
-            },
+            scale => $scale_json,
             scientist => undef,
-            status => $obsolete = 0 ? "Obsolete" : "Active",
+            status => "Active", # creating so will always be active
```

Comment:
Do we want this to always be active? I wonder if we will have the possibility of a user adding a legacy trait and wanting it to be set to 'Obsolete' from the start?

Reply:
ya, possibly, just didn't see it as a high priority and would have to add it to the saving side (not difficult) but wanted to get this wrapped up. Created a card in case we want to do it at some point: https://breedinginsight.atlassian.net/browse/BI-708?atlOrigin=eyJpIjoiYjk5YjQyNjkzMjZlNDIyNDkzZjY0OWVmYTJiNjUxMWMiLCJwIjoiaiJ9